### PR TITLE
Fix GRUCellBlockOp message for invalid rank of x

### LINKED ISSUE #58261: A Check Fail can be triggerred in GRUBlockCell
### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.9 and 2.12.0-dev20221018

### Custom Code

No

### OS Platform and Distribution

Linux Ubuntu 20.04

### Mobile device

_No response_

### Python version

3.8

### Bazel version

_No response_

### GCC/Compiler version

_No response_

### CUDA/cuDNN version

CUDA 11.5

### GPU model and memory

_No response_

### Current Behaviour?

```shell
A check fail can be triggerred in GRUBlockCell, which can lead to a crash.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import numpy as np
print(tf.__version__)
for _ in range(20):
    try:
            x = tf.random.uniform([1, 0, 1], dtype=tf.float32)
            h_prev = tf.random.uniform([1, 1, 1], dtype=tf.float32)
            w_ru = tf.random.uniform([1, 2, 1, 1, 1, 1], dtype=tf.float32)
            w_c = tf.random.uniform([1, 1, 1], dtype=tf.float32)
            b_ru = tf.random.uniform([2], dtype=tf.float32)
            b_c = tf.random.uniform([1], dtype=tf.float32)
            res = tf.raw_ops.GRUBlockCell(
                x=x,
                h_prev=h_prev,
                w_ru=w_ru,
                w_c=w_c,
                b_ru=b_ru,
                b_c=b_c,
            )
    except:
        pass
```

### Relevant log output

```shell
F tensorflow/core/framework/tensor_shape.cc:45] Check failed: NDIMS == dims() (2 vs. 3)Asking for tensor of 2 dimensions from a tensor of 3 dimensions
Aborted (core dumped)
```

### PR DESCRIPTION
**Pull request by nrwahl2**
The validation checks that `x` is a matrix, so rank must be 2. ff45913 fixed the crash in https://github.com/tensorflow/tensorflow/issues/58261 but left this typo in an exception message.

Fixes https://github.com/tensorflow/tensorflow/issues/58261

@mihaimaruseac

### CODE REVIEW

File: tensorflow/core/kernels/rnn/gru_ops.cc [hunk 1]
```diff
@@ -16,8 +16,9 @@ limitations under the License.
 #define EIGEN_USE_THREADS
 
 #include "tensorflow/core/kernels/rnn/gru_ops.h"
-#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+
```
Comment (mihaimaruseac) [inline tensorflow/core/kernels/rnn/gru_ops.cc:19]:
```suggestion
#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
```

File: tensorflow/core/kernels/rnn/gru_ops.cc [hunk 2]
```diff
@@ -16,8 +16,9 @@ limitations under the License.
...
 #include "tensorflow/core/framework/op_kernel.h"
+#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
```
Comment (mihaimaruseac) [inline tensorflow/core/kernels/rnn/gru_ops.cc:21]:
```suggestion
```

**Review by mihaimaruseac — COMMENTED** [review]
LGTM, just reordering the imports back as this would break Copybara otherwise

**Review by mihaimaruseac — APPROVED** [review]

**Comment by nrwahl2** [comment]
> LGTM, just reordering the imports back as this would break Copybara otherwise

No problem. Why's that?

**Comment by mihaimaruseac** [comment]
Internally the imports get rewritten based on where the code is situated in the Google monorepo and the type of the code (OSS -> third_party). Plus, code linters want the imports sorted.

**Pull request merged**